x/profiles/types/models_chain_links.go

```diff
@@ -61,16 +69,11 @@ func (p Proof) Validate() error {
 		return fmt.Errorf("public key field cannot be nil")
 	}
 
-	_, err := hex.DecodeString(p.Signature)
-	if err != nil {
-		return fmt.Errorf("invalid hex-encoded signature")
-	}
```

Comment:
Probably we should add a `Validate` method to the `SignatureData` interface so that we can call it here and make sure the signature is valid